### PR TITLE
Another monstrous pull.

### PR DESCRIPTION
**Pull request by Axiematic**

### CODE REVIEW

File: content/training/docker.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ requirements: |
   - Create accounts on the following services:
     - [AWS](http://aws.amazon.com/) and
     - [Docker Hub](https://hub.docker.com/).
-  - The [UNIX Essentials]({{< ref "unix-essentials.md" >}}) course is useful for understanding some of the details of a Dockerfile.
+  - The [UNIX Essentials](https://www.exegetic.biz/training/unix-essentials/) course is useful for understanding some of the details of a Dockerfile.
```
Comment (datawookie) [inline content/training/docker.md]: We can use the Hugo shorthand here for creating the link. See [this](https://gohugo.io/content-management/cross-references/).

File: content/training/r-machine-learning.md [hunk 2]
```diff
@@ -12,19 +12,18 @@ objectives: |
 outcomes: |
   Participants will be able to build Classification and Regression models on real world data. They will understand how the models work and how to interpret model predictions.
 requirements: |
-  Participants are assumed to have prior exposure to R, or at least to programming of some variety. Ideally participants should have completed the [Data Wrangling]({{< ref "r-data-wrangling.md" >}}) and [Visualisation]({{< ref "r-visualisation.md" >}}) modules.
+  Participants are assumed to have prior exposure to R, or at least to programming of some variety. Ideally participants should have completed the [Data Wrangling](https://www.exegetic.biz/training/r-data-wrangling/) and [Visualisation](https://www.exegetic.biz/training/r-visualisation/) modules.
```
Comment (datawookie) [inline content/training/r-machine-learning.md]: Definitely better to use shortcodes here too, please!

File: content/training/r-machine-learning-time-series.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ objectives: |
 outcomes: |
 
 requirements: |
-  We assume that participants have prior experience with R, ideally having completed both the the [Introduction to R]({{< ref "r-introduction.md" >}}) and [Data Wrangling]({{< ref "r-data-wrangling.md" >}}) courses.
+  We assume that participants have prior experience with R, ideally having completed both the the [Introduction to R](https://www.exegetic.biz/training/r-introduction/) and [Data Wrangling](https://www.exegetic.biz/training/r-data-wrangling/) courses.
```
Comment (datawookie) [inline content/training/r-machine-learning-time-series.md]: Use shortcodes.

File: content/training/r-markdown-automated-reporting.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ who: The course is aimed at students, academics and professionals who need to ge
 objectives: Reports are an important artefact of research and analysis. Often these reports need to be frequently generated or updated. Ideally this should be done in a reproducible manner. It's is an ideal opportunity for automation! In this course you'll learn how to use R and Markdown to efficiently generate sophisticated and visually appealing reports.
 outcomes: Participants will be able to generate reports which include text, tables, figures, code and analytical results. They will also be able to flexibly style the reports.
 setup: |
-  - Participants are assumed to have prior exposure to R, or at least to programming of some variety. Ideally participants should have completed the [Data Wrangling]({{< ref "r-data-wrangling.md" >}}) and [Visualisation]({{< ref "r-visualisation.md" >}}) modules.
+  - Participants are assumed to have prior exposure to R, or at least to programming of some variety. Ideally participants should have completed the [Data Wrangling](https://www.exegetic.biz/training/r-data-wrangling/) and [Visualisation](https://www.exegetic.biz/training/r-visualisation/) modules.
```
Comment (datawookie) [inline content/training/r-markdown-automated-reporting.md]: Use shortcodes.

File: content/training/r-shiny-apps.md [hunk 5]
```diff
@@ -19,7 +19,7 @@ outcomes: |
   - understand reactivity and how the UI and server communicate with each other; and
   - know how to deploy a Shiny application.
 requirements: |
-  Participants are assumed to have prior exposure to R, or at least to programming of some variety. Ideally participants should have completed the [Data Wrangling]({{< ref "r-data-wrangling.md" >}}) and [Visualisation]({{< ref "r-visualisation.md" >}}) modules.
+  Participants are assumed to have prior exposure to R, or at least to programming of some variety. Ideally participants should have completed the [Data Wrangling](https://www.exegetic.biz/training/r-data-wrangling/) and [Visualisation](https://www.exegetic.biz/training/r-visualisation/) modules.
```
Comment (datawookie) [inline content/training/r-shiny-apps.md]: Use shortcodes.

File: content/training/r-visualisation.md [hunk 6]
```diff
@@ -14,7 +14,7 @@ who: The course is aimed at students, academics and professionals who need to vi
 objectives: Being able to communicate the results of an analysis is as important as doing the analysis itself. In this course you'll learn how to use two R packages (ggplot2 and plotly) to create effective, visually pleasing and reproducible visualisations.
 outcomes: Participants will be able to create static and animated visualisations with ggplot2 and interactive visualisations using plotly. They will also be able to customise the presentation characteristics of these visualisations.
 requirements: |
-  Participants are assumed to have prior experience with R, ideally having completed both the the [Introduction to R]({{< ref "r-introduction.md" >}}) and [Data Wrangling]({{< ref "r-data-wrangling.md" >}}) courses.
+  Participants are assumed to have prior experience with R, ideally having completed both the the [Introduction to R](https://www.exegetic.biz/training/r-introduction/) and [Data Wrangling](https://www.exegetic.biz/training/r-data-wrangling/) courses.
```
Comment (datawookie) [inline content/training/r-visualisation.md]: Use shortcodes.

**Review by datawookie — CHANGES_REQUESTED** [review]
Hi @Axiematic, this looks pretty good. I see that there's a lot of content added to .../old-training-courses/. I've made a card to clean that up. Also noted a few cases where it would be better to use Hugo shortcodes for links rather than hard-coding the links. Thanks, ABC/

**Comment by datawookie** [comment]
Hi @Axiematic,

Okay, whew! This is huge. I've taken another look and I'm generally happy with this. Some parts of this really need to go live ASAP. But...

1. There are some change requests that have not been dealt with (all relating to using Hugo shortcodes for links rather than hard-coded paths).
2. Let's try not to let this happen again. Rather make a pull request after every feature has been completed. So do something, check that it works/looks good and then immediately create a pull request.

Please let's sort this out ASAP on Tuesday.

Thanks,
Dad.

**Comment by Axiematic** [comment]
Hi @datawookie 

1. None of the shortcode methods seem to work in the front matter. I've done some extensive Googling and can't seem to find a solution other than just using the original links.

**Comment by datawookie** [comment]
Ah, shit. How annoying.

Okay, I'm going to merge this. Please check on the site tomorrow morning to ensure that nothing had broken.

And let's keep these PR short and sweet in future.

**Review by datawookie — APPROVED** [review]

**Comment by Axiematic** [comment]
@datawookie everything looks good to me :).

**Pull request merged**